backend/infrahub/core/validators/node/relationship.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+from ..interface import ConstraintCheckerInterface
+from ..query import NodeNotPresentValidatorQuery
+
+if TYPE_CHECKING:
+    from infrahub.core.branch import Branch
+    from infrahub.core.path import GroupedDataPaths
+    from infrahub.database import InfrahubDatabase
+
+    from ..model import SchemaConstraintValidatorRequest
+
+
+class NodeRelationshipAddChecker(ConstraintCheckerInterface):
+    query_classes = [NodeNotPresentValidatorQuery]
```

Comment:
is this the right query for this constraint? it looks like `NodeNotPresentValidatorQuery` only looks at attributes, not relationships

Reply:
Good point, the goal of the query is to focus just on the node, not specifically on the attributes or the relationships, I will clean it up